sdk/src/transaction.rs

```diff
@@ -204,16 +217,29 @@ impl Transaction {
                 .for_each(|signature| *signature = Signature::default());
         }
 
-        for keypair in keypairs {
-            let i = signed_keys
-                .iter()
-                .position(|pubkey| pubkey == &keypair.pubkey())
-                .expect("keypair-pubkey mismatch");
-
-            self.signatures[i] = keypair.sign_message(&self.message_data())
+        for i in 0..positions.len() {
+            self.signatures[positions[i]] = keypairs[i].sign_message(&self.message_data())
         }
     }
 
+    /// Get the positions of the pubkeys in `account_keys` associated with signing keypairs
+    pub fn get_signing_keypair_positions<T: KeypairUtil>(
+        &self,
+        keypairs: &[&T],
+    ) -> Vec<Option<usize>> {
+        let signed_keys =
+            &self.message.account_keys[0..self.message.header.num_required_signatures as usize];
```

Comment:
This will panic if `num_required_signatures > account_keys`. Will this cause a problem for  `extern "C" fn transaction_partial_sign`?

Reply:
Oops, good catch. The whole point of breaking this out was to avoid panics in the `extern "C"` functions.

Reply:
:) Thought so